storage-operator/cmd/manager/main.go

```diff
@@ -79,14 +82,18 @@ func main() {
 		os.Exit(1)
 	}
 
+	ctx := context.TODO()
+	// Become the leader before proceeding
+	err = leader.Become(ctx, "storage-operator-lock")
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
 	// Create a new Cmd to provide shared dependencies and start components
 	mgr, err := manager.New(cfg, manager.Options{
-		Namespace:               namespace,
-		MapperProvider:          restmapper.NewDynamicRESTMapper,
-		MetricsBindAddress:      fmt.Sprintf("%s:%d", metricsHost, metricsPort),
-		LeaderElection:          true,
-		LeaderElectionNamespace: "kube-system",
-		LeaderElectionID:        "storage-operator-lock",
+		Namespace:          namespace,
+		MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
```

Comment:
IIRC this is reverting bff55c256fff5dff9525f86bd0b6b7dfbc4735b7

If this is done on purpose, this should be at least justified in the commit message.
If this an error, let's fix it :wink: